src/main/java/mate/academy/mybookshop/controller/CategoryController.java

```diff
@@ -0,0 +1,75 @@
+import mate.academy.mybookshop.dto.CategoryResponseDto;
+import mate.academy.mybookshop.dto.CreateCategoryRequestDto;
+import mate.academy.mybookshop.dto.UpdateCategoryRequestDto;
+import mate.academy.mybookshop.service.BookService;
+import mate.academy.mybookshop.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for managing books categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @GetMapping
+    @Operation(summary = "Get all categories", description = "Get a list of the all categories")
+    public List<CategoryResponseDto> findAll(Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @GetMapping("/{id}")
+    @Operation(summary = "Find category by id",
+            description = "Get one category with request id from all available categories")
+    public CategoryResponseDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getCategoryById(id);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    @Operation(summary = "Create a new category", description = "Create a new category")
+    public CategoryResponseDto create(@RequestBody @Valid CreateCategoryRequestDto requestDto) {
+        return categoryService.create(requestDto);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping("/{id}")
+    @Operation(summary = "Update category info",
+            description = "Update info about exist category by id")
+    public CategoryResponseDto update(@PathVariable Long id,
+                                          @RequestBody @Valid UpdateCategoryRequestDto requestDto) {
+        return categoryService.update(id, requestDto);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @DeleteMapping("/{id}")
+    @Operation(summary = "Delete category", description = "Delete category from DB by id")
+    public void delete(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @GetMapping("/{id}/books")
+    @Operation(summary = "Get books by category id",
+            description = "Get all books by category id")
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(@PathVariable Long id) {
```

Comment:
I think there should be `Pageable` parameter too

Reply:
thx, added